server/src/api/install-nodejs.ts

```diff
@@ -0,0 +1,82 @@
+import { execSync, exec } from 'child_process';
+import * as fs from 'fs';
+import * as request from 'request';
+import { platform } from 'os';
+import { exists } from '../utils';
+import * as progress from 'progress-stream';
+import { InstallNodeJsStatus } from '../graphql-types';
+
+export interface Progress extends fs.WriteStream {
+  progress(): {
+    percentage: number;
+    transferred: number;
+    length: number;
+    remaining: number;
+    eta: number;
+    runtime: number;
+    delta: number;
+    speed: number;
+  };
+}
+
+export let nodeDownloadProgress: Progress | null = null;
+export let nodeDownload: fs.WriteStream | null = null;
+export let nodeInstallDone: boolean = true;
+
+export function installNodeJs(): InstallNodeJsStatus {
+  nodeDownloadProgress = null;
+  nodeDownload = null;
+  nodeInstallDone = false;
+  try {
+    switch (platform()) {
+      case 'darwin':
+        nodeDownloadProgress = progress({
+          length: 15546624,
+          time: 100
+        }) as Progress;
+        const pkg = fs.createWriteStream('node.pkg').on('finish', () => {
```

Comment:
you should use the tmp directory. you don't want to write it into the current directory

Reply:
You can use `import { dirSync } from 'tmp';`